src/Raven.Server/Documents/ETL/EtlLoader.cs

```diff
@@ -154,11 +153,15 @@ private void NotifyAboutWork(DocumentChange documentChange)
             }
         }
 
-        private void HandleSystemDocumentChange(DocumentChange change)
+        public virtual void Dispose()
         {
-            if (change.Id.Equals(Constants.Documents.ETL.RavenEtlDocument, StringComparison.OrdinalIgnoreCase) == false)
-                return;
+            _database.Changes.OnDocumentChange -= NotifyAboutWork;
 
+            Parallel.ForEach(_processes, x => x.Dispose());
```

Comment:
Can the `Dispose` throw here? If so, we need to make sure that we dispose all them.